fido/Crypto/Fido2/Model/JavaScript/Decoding.hs

```diff
@@ -277,6 +284,123 @@ instance Decode (M.PublicKeyCredential 'M.Get) where
     pkcClientExtensionResults <- decode clientExtensionResults
     pure $ M.PublicKeyCredential {..}
 
+instance Decode M.RpId
+
+instance Decode M.RelyingPartyName
+
+instance Decode M.PublicKeyCredentialRpEntity where
+  decode JS.PublicKeyCredentialRpEntity {..} = do
+    pkcreId <- decode id
+    pkcreName <- decode name
+    pure $ M.PublicKeyCredentialRpEntity {..}
+
+instance Decode M.UserAccountDisplayName
+
+instance Decode M.UserAccountName
+
+instance Decode M.PublicKeyCredentialUserEntity where
+  decode JS.PublicKeyCredentialUserEntity {..} = do
+    pkcueId <- decode id
+    pkcueDisplayName <- decode displayName
+    pkcueName <- decode name
+    pure $ M.PublicKeyCredentialUserEntity {..}
+
+instance Decode M.Challenge
+
+instance Decode PublicKey.COSEAlgorithmIdentifier where
+  decode n = maybe (Left $ DecodingErrorUnexpectedAlgorithmIdentifier n) Right $ PublicKey.toAlg n
+
+instance Decode M.Timeout
+
+instance Decode [M.AuthenticatorTransport] where
+  decode = pure . mapMaybe decodeTransport
+    where
+      decodeTransport "usb" = Just M.AuthenticatorTransportUSB
+      decodeTransport "nfc" = Just M.AuthenticatorTransportNFC
+      decodeTransport "ble" = Just M.AuthenticatorTransportBLE
+      decodeTransport "internal" = Just M.AuthenticatorTransportInternal
+      decodeTransport _ = Nothing
+
+instance Decode [M.PublicKeyCredentialDescriptor] where
+  decode Nothing = pure []
+  decode (Just xs) = catMaybes <$> traverse decodeDescriptor xs
+    where
+      decodeDescriptor :: JS.PublicKeyCredentialDescriptor -> Either DecodingError (Maybe M.PublicKeyCredentialDescriptor)
+      decodeDescriptor JS.PublicKeyCredentialDescriptor {typ = "public-key", id, transports} = do
+        let pkcdTyp = M.PublicKeyCredentialTypePublicKey
+        pkcdId <- decode id
+        pkcdTransports <- decode transports
+        pure . Just $ M.PublicKeyCredentialDescriptor {..}
+      decodeDescriptor _ = pure Nothing
+
+instance Decode M.AuthenticatorAttachment where
+  decode "platform" = Right M.AuthenticatorAttachmentPlatform
+  decode "cross-platform" = Right M.AuthenticatorAttachmentCrossPlatform
+  decode attachment = Left $ DecodingErrorUnexpectedAttachment attachment
```

Comment:
This shouldn't throw an error, see https://www.w3.org/TR/webauthn-2/#dom-authenticatorselectioncriteria-authenticatorattachment